director/template/var_kv.go

```diff
@@ -33,7 +33,11 @@ func (a *VarKV) UnmarshalFlag(data string) error {
 		return bosherr.WrapErrorf(err, "Deserializing variables '%s'", data)
 	}
 
-	*a = VarKV{Name: pieces[0], Value: vars}
+	if _, ok := vars.(string); ok {
```

Comment:
I'm pretty bad at Go, but this is weird enough that it should have a comment explaining what it's doing and why it's doing it. I can _guess_, but I definitely wouldn't be confident in my guess without a lot of research.

Reply:
I think the technical term for this is a [type assertion](https://go.dev/tour/methods/15)

Reply:
sure, the commit message serves as an explanation, but a comment can be added as a convenience.

Reply:
_This_ must be a lie, right?

> If i does not hold a T, the statement will trigger a panic.